### ISSUE TITLE
python 3.11 branch was force pushed ?

### ISSUE THREAD

**Issue by asottile** [python/cpython#92478]
**Bug report**

I have nightly jobs which pull cpython and build packages for ubuntu such that users can try them out early on: https://github.com/deadsnakes/nightly

today the python3.11 runs are failing due to the pull from `cpython` being non-fast-forward

the 3.11 branch (though it was named `main` at the time) contained this commit which has been rebased (?): 0924b95f6e678beaf4a059d679515956bac608fb

the new corresponding commit appears to be the roughly the same, though with slightly different attribution:

```diff
$ diff -u <(git show 565a5a32962c16aa7d494002b625d308d044a72f ) <(git show 0924b95f6e678beaf4a059d679515956bac608fb)
--- /dev/fd/63	2022-05-08 07:37:27.697601140 -0700
+++ /dev/fd/62	2022-05-08 07:37:27.701603140 -0700
@@ -1,9 +1,8 @@
-commit 565a5a32962c16aa7d494002b625d308d044a72f
-Author: Miss Islington (bot) <31488909+miss-islington@users.noreply.github.com>
-Date:   Sat May 7 21:04:04 2022 -0700
+commit 0924b95f6e678beaf4a059d679515956bac608fb
+Author: Erlend Egeberg Aasland <erlend.aasland@innova.no>
+Date:   Sat May 7 06:23:13 2022 +0200
 
-    gh-92308: Add Pending Removal section to 3.11 What's New (GH-92309)
-    
+    gh-92308: Add Pending Removal section to 3.11 What's New (#92309)
     
     * gh-92308: Add Pending Removal section to 3.11 What's New
     
@@ -25,9 +24,6 @@
     
     Co-authored-by: Victor Stinner <vstinner@python.org>
     Co-authored-by: Ezio Melotti <ezio.melotti@gmail.com>
-    (cherry picked from commit 0924b95f6e678beaf4a059d679515956bac608fb)
-    
-    Co-authored-by: Erlend Egeberg Aasland <erlend.aasland@innova.no>
 
 diff --git a/Doc/whatsnew/3.11.rst b/Doc/whatsnew/3.11.rst
 index b9871f7ef6..c4e8e6f9a1 100644
``` 

here is the fast-forward pull which originally brought in the commit which is now gone: https://github.com/deadsnakes/python3.11-nightly/runs/6333419745?check_suite_focus=true

here's an example failing run which can no longer fast-forward pull cpython/3.11: https://github.com/deadsnakes/python3.11-nightly/actions/runs/2289092862

A few questions
- was this intentional?
- is there some workflow that needs to be adjusted?
- should I force push as well?

**Your environment**

N/A

**Comment by pablogsal** [comment]
> * was this intentional?

Yes. The problem is that I blocked the main branch so no commits were pushed during the beta 1 release and branch changes. But some core devs that were not aware of the restriction seem to have overridden my block using admin privileges and pushed a commit (https://github.com/python/cpython/pull/92309). This should not have happened as I was still engineering the 3.11 branch. I had to manually rebase the commit directly from the original one and force pushed to the 3.11 branch to fix some conflicts with the branch and the workflow.

> * is there some workflow that needs to be adjusted?

Not really. This was just some one-off problem.

> should I force push as well?

Yes, the current state of 3.11 is the canonical one
